### DISCUSSION TITLE
-attenuate doesn't do what I expect

### DISCUSSION THREAD

**Discussion by jonathancolledge** [ImageMagick/ImageMagick#7088] [Help]
If I use this

````
-attenuate 0.1 +noise poisson
````
I get complete noise

If I use anything below 1 the whole image is noise. 

If I use numbers greater than 1, I get a lot of noise. But, 

```
-attenuate 99 +noise poisson
```
Results in quite a bit less noise, but still far too much.

Much greater values result in a black image.

What on earth should I put in for a tiny bit of added noise please? The default is 1.0 for maximum noise, so I had thought that <1 would be less noise, but that does not appear to be the case.

(imagemagick 7.1.1-27 hdri q16 on Windows 10)

**Comment by fmw42** [comment]
I get correct results when testing poisson noise.  As you specify a smaller attenuation (less than 1), you will get a darker image.  That is because poisson noise is not additive.  See https://tomroelandts.com/articles/gaussian-noise-is-added-poisson-noise-is-applied

See examples at 
https://en.wikipedia.org/wiki/File:Photon-noise.jpg which is the top figure at https://en.wikipedia.org/wiki/Shot_noise

Perhaps you should try Gaussian noise. It behaves the way you want (additive) and you will just see less noise as the attenuation is lowered, not a darkening.

**Reply by jonathancolledge** [reply]
With a small number I get a brighter image with all noise.

I only get a dark image with a number like 250.

I was after something like the middle bottom of the Wikipedia image with signal dependent noise.

**Comment by fmw42** [comment]
There is no level that satisfies your request.  Poisson noise does not behave that way.  But if you want to make an image that uses poisson noise as additive, then you need to create a noise only image and add it to your normal image.

**Reply by jonathancolledge** [reply]
Ah, thank you. I get it. Imagemagick is generating Poisson noise from my image, not simulating an imaging system imaging my image. That makes sense how it is working.

**Reply by jonathancolledge** [reply]
I've just come across this for film grain
http://www.fmwconcepts.com/imagemagick/filmgrain/index.php

In the examples, the arguments:
-a 100 -A 0 -n poisson

But you can't do that. The 50% Gray has removed the signal dependence so it isn't really going to be Poisson noise related to imaging the image/scene that you are adding the noise to.

**Reply by fmw42** [reply]
In my script, I allow any type of noise, though, poisson is not really additive.  It does not really make sense to use Poisson distribution.  But I let the user do that if they really want.  The default for the script is Gaussian.  (Large poisson values are similar to Gaussian, though not for small values).
